apps/judicial-system/backend/src/app/modules/case-table/whereOptions/prosecutor.ts

```diff
@@ -0,0 +1,101 @@
+import { fn, Op } from 'sequelize'
+
+import {
+  CaseAppealState,
+  CaseState,
+  completedIndictmentCaseStates,
+  completedRequestCaseStates,
+  indictmentCases,
+  investigationCases,
+  restrictionCases,
+  type User as TUser,
+} from '@island.is/judicial-system/types'
+
+const prosecutorSharedWhereOptions = (user: TUser) => ({
+  [Op.or]: [
+    { creating_prosecutor_id: user.id },
+    { prosecutor_id: user.id },
+    { prosecutors_office_id: user.institution?.id },
+    { shared_with_prosecutors_office_id: user.institution?.id },
+  ],
+})
+
+// Prosecutor Request Cases
+
+const prosecutorRequestCasesSharedWhereOptions = (user: TUser) => ({
+  is_archived: false,
+  type: [...restrictionCases, ...investigationCases],
+  is_heightened_security_level: { [Op.or]: [null, false] },
+  ...prosecutorSharedWhereOptions(user),
+})
+
+export const prosecutorRequestCasesInProgressWhereOptions = (user: TUser) => ({
+  ...prosecutorRequestCasesSharedWhereOptions(user),
+  state: [CaseState.DRAFT, CaseState.SUBMITTED, CaseState.RECEIVED],
+})
+
+export const prosecutorRequestCasesActiveWhereOptions = (user: TUser) => ({
+  [Op.and]: [
+    prosecutorRequestCasesSharedWhereOptions(user),
+    {
+      [Op.or]: [
+        {
+          initial_ruling_date: {
```

Comment:
Note that checking the ruling date is not necessary. If the case is accepted, then it should be shown in this list as long as the valid to date is in the future. Please remove these checks.